services/weblate/weblate-project-translated-percentage.service.js

```diff
@@ -0,0 +1,72 @@
+'use strict'
+
+const Joi = require('joi')
+const { BaseJsonService } = require('..')
+const { optionalUrl } = require('../validators')
+const { colorScale } = require('../color-formatters')
+
+const schema = Joi.object({
+  translated_percent: Joi.number().required(),
+}).required()
+
+const queryParamSchema = Joi.object({
+  server: optionalUrl.required(),
+}).required()
+
+const documentation = `
+  <p>
+    This badge displays the percentage of strings translated on a project on a
+    Weblate instance.
+  </p>
+`
+
+module.exports = class WeblateProjectTranslatedPercentage extends (
+  BaseJsonService
+) {
+  static category = 'other'
+  static route = { base: 'weblate', pattern: ':project', queryParamSchema }
+
+  static examples = [
+    {
+      title: 'Weblate project translated',
+      namedParams: { project: 'godot-engine' },
+      queryParams: { server: 'https://hosted.weblate.org' },
+      staticPreview: this.render({ translatedPercent: 20.5 }),
+      documentation,
+      keywords: ['i18n', 'translation', 'internationalization'],
+    },
+  ]
+
+  static defaultBadgeData = { label: 'translated' }
+
+  /**
+   * Takes a percentage and maps it to a message and color.
+   *
+   * The colors are determined based on how Weblate does it internally.
+   * {@link https://github.com/WeblateOrg/weblate/blob/main/weblate/trans/widgets.py Weblate on GitHub}
+   *
+   * @param {*} translatedPercent The percentage of translations translated.
+   * @returns {object} Format for the badge.
+   */
```

Comment:
hmm, i'm a little torn on this. We don't typically use the full-blown jsdoc on functions except where needed (and do have linter checks for it). I'm not necessarily opposed to this, but also a bit wary of starting a pattern of applying jsdoc to the standard functions that exist in every single one of our service classes. For now I'd prefer just adding an line comment to the `colorScale` call within the function (basically copy/combine lines 45/46 and move that to apply to line 52), but cc @chris48s in case you feel differently since you did all the original jsdoc work

Reply:
We have acquired a couple of random services where we have got JSDoc comments like this

https://contributing.shields.io/module-services_dynamic_json-path.html
https://contributing.shields.io/module-services_steam_steam-base-BaseSteamAPI.html

but I think you're probably right that the generated docs are only really useful as a reference for functions/classes which we are going to use in more than one place. I'm not going to use `WeblateProjectTranslatedPercentage.render()` anywhere else so its probably just noise to include it in the docs site (esp if we had hundreds of these). I don't want to discourage contributors from including useful information in comments for service classes though. Maybe we could add en exclude pattern to [jsdoc.json](https://github.com/badges/shields/blob/master/jsdoc.json) to ignore `services/**/*.service.js` files when we build?
Either way, I don't think it has to be a blocker for this PR.

Reply:
> Either way, I don't think it has to be a blocker for this PR.

Agreed. My concern was really about the hypothetical scenario of this being repeated at scale (every render, handle, etc. function in every service class) and the impact that'd have on code scannability and the docs site.

> Maybe we could add en exclude pattern to jsdoc.json to ignore services/**/*.service.js files when we build?

I like this idea :+1: (and feel free to tackle that one @SethFalco if you're looking for anything else after this!)